### PR TITLE
EventToCommand Behavior Sample implementation.

### PR DESCRIPTION
**Pull request by almirvuk**
[[Update] 14 EventToCommandBehavior #56](https://github.com/PrismLibrary/Prism-Samples-Forms/issues/56)

I have implemented this EventToCommand Behaviour sample using three standard examples. 
 
Page(s): 
* EventArgsConverterExamplePage
* SimpleExamplePage
* EventArgsParameterExamplePage

Model(s) (test data model for showing data in the list):
* Developer 

Converter(s):
* ItemTappedEventArgsConverter

This project sample is showing how to use EventToCommand Behavior in three simple examples, on pages with `ListView `and SelectedItem commands in the `ViewModel `part.

Let me know if there is a need to change and edit something.

### CODE REVIEW

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="PrismSample.Views.EventArgsConverterExamplePage"
+             Title="Event Args Converter Example"
+             
+             xmlns:c="clr-namespace:PrismSample.Converters"
+             xmlns:b="clr-namespace:Prism.Behaviors;assembly=Prism.Forms">
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml]: this should be `xmlns:prism="http://prismlibrary.com"`

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="PrismSample.Views.EventArgsParameterExamplePage"
+             Title="Event Args Example"
+             
+             xmlns:b="clr-namespace:Prism.Behaviors;assembly=Prism.Forms">
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml]: this should be `xmlns:prism="http://prismlibrary.com"`

File: 14-EventToCommandBehavior/src/PrismSample/Views/MainPage.xaml [hunk 3]
```diff
@@ -5,12 +5,21 @@
              xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
              mc:Ignorable="d"
              Title="Main Page"
-             x:Class="PrismSample.Views.MainPage">
+             x:Class="PrismSample.Views.MainPage"
+  
+             xmlns:prism="clr-namespace:Prism.Navigation.Xaml;assembly=Prism.Forms">
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/MainPage.xaml]: this should be `xmlns:prism="http://prismlibrary.com"`

File: 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="PrismSample.Views.SimpleExamplePage"
+             Title="Simple usage"
+             
+             xmlns:b="clr-namespace:Prism.Behaviors;assembly=Prism.Forms">
+
+  <ContentPage.Content>
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml]: there's no need for ContentPage.Content

File: 14-EventToCommandBehavior/src/PrismSample/Views/MainPage.xaml [hunk 5]
```diff
@@ -5,12 +5,21 @@
...
+  <StackLayout Margin="16" VerticalOptions="Center">
+
+    <Button Text="Simple Example" 
+            Command="{prism:NavigateTo 'NavigationPage/SimpleExamplePage'}"/>
+
+    <Button Text="IValueConverter Example" 
+            Command="{prism:NavigateTo 'NavigationPage/EventArgsConverterExamplePage'}"/>
+
+    <Button Text="Event Args Parameter Path Example" 
+            Command="{prism:NavigateTo 'NavigationPage/EventArgsParameterExamplePage'}"/>
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/MainPage.xaml]: use a traditional command with Navigation in the ViewModel. This is a sample and if you're going through the sample with EventToCommand we've introduced conventional navigation but not XAML navigation by the time you get here.

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml (hunk 1 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml]: I will fix it :)

File: 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml (hunk 4 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml]: I will fix it :)

File: 14-EventToCommandBehavior/src/PrismSample/Views/MainPage.xaml (hunk 5 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/Views/MainPage.xaml]: In the first "edition" of the sample, I used that approach, and later on, before the commit, I decided to change the traditional way with this "shorter" way. Will fix it tonight!

**Comment by almirvuk** [comment]
@dansiegel I have made some changes, can you please take a look :) Thanks!

File: 14-EventToCommandBehavior/src/PrismSample/ViewModels/EventArgsConverterExamplePageViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+{
+    public class EventArgsConverterExamplePageViewModel : BindableBase
+    {
+        private readonly IPageDialogService _pageDialogService;
+
+        public ObservableCollection<Developer> Developers { get; set; }
+
+        private DelegateCommand<Developer> _selectedDeveloperCommand;
+        public DelegateCommand<Developer> SelectedDeveloperCommand => 
+            _selectedDeveloperCommand ?? (_selectedDeveloperCommand = new DelegateCommand<Developer>((developer) => SelectedDeveloper(developer)));
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/ViewModels/EventArgsConverterExamplePageViewModel.cs]: let's keep things consistent and clean... follow the same pattern for declaring your Commands as on the MainPageViewModel

File: 14-EventToCommandBehavior/src/PrismSample/ViewModels/EventArgsExamplePageViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+namespace PrismSample.ViewModels
+{
+    public class EventArgsExamplePageViewModel : BindableBase
+    {
+        private readonly IPageDialogService _pageDialogService;
+
+        public ObservableCollection<Developer> Developers { get; set; }
+
+        private DelegateCommand<Developer> _selectedDeveloperCommand;
+        public DelegateCommand<Developer> SelectedDeveloperCommand => _selectedDeveloperCommand ?? (_selectedDeveloperCommand = new DelegateCommand<Developer>((developer) => SelectedDeveloper(developer)));
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/ViewModels/EventArgsExamplePageViewModel.cs]: let's keep things consistent and clean... follow the same pattern for declaring your Commands as on the MainPageViewModel

File: 14-EventToCommandBehavior/src/PrismSample/ViewModels/MainPageViewModel.cs [hunk 8]
```diff
@@ -1,11 +1,41 @@
...
+
+        private async void GoToEventArgsParameterExamplePage()
+        {
+            await _navigationService.NavigateAsync(nameof(NavigationPage) + "/" + nameof(EventArgsParameterExamplePage));
+        }
+
+        private async void GoToSimpleExamplePage()
+        {
+            await _navigationService.NavigateAsync(nameof(NavigationPage) + "/" + nameof(SimpleExamplePage));
+        }
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/ViewModels/MainPageViewModel.cs]: This 100% breaks MVVM when you use `nameof(SomePage)` in a ViewModel. Also the navigation pattern here makes for a poor experience where you have to stopped the app and rerun to see another example... update these just to the name of the page you want and update the navigation in App.OnInitialized to use `NavigationPage/MainPage`

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="PrismSample.Views.EventArgsConverterExamplePage"
+             Title="Event Args Converter Example"
+
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml]: remove the empty line and whitespace here

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml [hunk 10]
```diff
@@ -0,0 +1,34 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="PrismSample.Views.EventArgsConverterExamplePage"
+             Title="Event Args Converter Example"
+             
+             xmlns:c="clr-namespace:PrismSample.Converters"
+             xmlns:b="http://prismlibrary.com">
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml]: any time you are referencing the `http://prismlibrary.com` namespace the namespace should always be `xmlns:prism="http://prismlibrary.com"`

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,14 @@
+﻿using Xamarin.Forms.Xaml;
+using Xamarin.Forms;
+
+namespace PrismSample.Views
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml.cs]: there is no need for this as we already turned it on at the assembly level

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+    <ListView.Behaviors>
+      <b:EventToCommandBehavior EventName="ItemTapped" 
+                                Command="{Binding SelectedDeveloperCommand}"
+                                EventArgsParameterPath="Item" />
+    </ListView.Behaviors>
+
+  </ListView>
+
+</ContentPage>
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml]: see comments for EventArgsConverterExamplePage

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml.cs [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class EventArgsParameterExamplePage : ContentPage
+    {
+        public EventArgsParameterExamplePage()
+        {
+            InitializeComponent();
+        }
+    }
+}
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml.cs]: see comments for EventArgsConverterExamplePage

File: 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+    </ListView.ItemTemplate>
+
+    <ListView.Behaviors>
+      <b:EventToCommandBehavior EventName="ItemTapped" 
+                                Command="{Binding SelectedDeveloperCommand}" />
+    </ListView.Behaviors>
+
+  </ListView>
+
+</ContentPage>
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml]: See comments for EventArgsConverterExamplePage

File: 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml.cs [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class SimpleExamplePage : ContentPage
+    {
+        public SimpleExamplePage()
+        {
+            InitializeComponent();
+        }
+    }
+}
```
Comment (dansiegel) [inline 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml.cs]: See comments for EventArgsConverterExamplePage

File: 14-EventToCommandBehavior/src/PrismSample/ViewModels/EventArgsConverterExamplePageViewModel.cs (hunk 6 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/ViewModels/EventArgsConverterExamplePageViewModel.cs]: Done with the latest commit. :)

File: 14-EventToCommandBehavior/src/PrismSample/ViewModels/EventArgsExamplePageViewModel.cs (hunk 7 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/ViewModels/EventArgsExamplePageViewModel.cs]: Done with the latest commit. :)

File: 14-EventToCommandBehavior/src/PrismSample/ViewModels/MainPageViewModel.cs (hunk 8 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/ViewModels/MainPageViewModel.cs]: Done with the latest commit. :)

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml (hunk 9 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml]: Done with the latest commit. :)

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml (hunk 10 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsConverterExamplePage.xaml]: Done with the latest commit. :)

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml (hunk 12 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml]: Done with the latest commit. :)

File: 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml.cs (hunk 13 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/Views/EventArgsParameterExamplePage.xaml.cs]: Done with the latest commit. :)

File: 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml (hunk 14 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml]: Done with the latest commit. :)

File: 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml.cs (hunk 15 above)
Comment (almirvuk) [reply 14-EventToCommandBehavior/src/PrismSample/Views/SimpleExamplePage.xaml.cs]: Done with the latest commit. :)

**Comment by almirvuk** [comment]
Hi @dansiegel, I have made some changes to the code files. Thank you for your lastest review :)

**Comment by dansiegel** [comment]
Looks great, thanks for the PR

**Pull request merged**